andela/technites-bn-backend
# #168049997 An Email should be sent to a user upon successful registration. User should be able to receive a verification link via email upon successful signup

#### What does this PR do?
- An email should be sent to a user upon successful registration.
#### Description of Task to be completed?
- Create a SendGrid account and create an API KEY
- Send a verification email to a user once he/she finishes to signup
- Activate an account as verified once a user confirms his/her email
#### How should this be manually tested?
1. Run `git checkout ft-email-verification-168049997`  to your terminal.

2. Run `npm install`

3. Remember to add the variable `SENDGRID_API_KEY` to your `.env` file with the corresponding value.

4. Run `npm run migrate:refresh` to create users tables and models

5. Run `npm run dev`

6. Sign up a user using this endpoint `/api/v1/auth/register` using `POST` method, you will also need to add in an `email`, `password`, `firstname`, `lastname` , `username` fields

7. After sign up, you'll receive a confirmation email and then click on the `confirm email` button

8. You will receive a response after it is complete

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168049997](https://www.pivotaltracker.com/story/show/168049997)
#### What are the relevant Github Issues?
N/A
#### Screenshots (if appropriate)
*Register a user*
<img width="1116" alt="Screen Shot 2019-09-12 at 10 43 27" src="https://user-images.githubusercontent.com/33954234/64768697-33828a00-d54a-11e9-9779-7323e43de7f7.png">

*Email verification*
<img width="1156" alt="Screen Shot 2019-09-12 at 10 42 49" src="https://user-images.githubusercontent.com/33954234/64768722-3c735b80-d54a-11e9-9cac-b14e0385e2d5.png">


N/A
#### Questions:
N/A
